faker/providers/address/fr_CH/__init__.py

```diff
@@ -123,6 +123,8 @@ def canton_name(self):
         """
         return self.canton()[1]
 
+    administrative_unit = canton_name
```

Comment:
`administrative_unit` should be the method with logic in, and then we should have `canton_name = administrative_unit`